src/dataflow-types/src/client.rs

```diff
@@ -961,28 +1018,199 @@ pub mod process_local {
+                        }
+                    }
+                    GrpcComputedTcpConn::Connected(_) => break,
+                    GrpcComputedTcpConn::Backoff(deadline) => {
+                        time::sleep_until(*deadline).await;
+                        self.state = GrpcComputedTcpConn::Disconnected;
+                    }
+                }
+            }
+        }
+    }
+
+    #[async_trait]
+    impl<T> GenericClient<ComputeCommand<T>, ComputeResponse<T>> for GrpcComputedClient
+    where
+        T: ProtoSerializableTimestamp + 'static + std::fmt::Debug + Send,
+    {
+        async fn send(&mut self, cmd: ComputeCommand<T>) -> Result<(), anyhow::Error> {
+            let sender = if let GrpcComputedTcpConn::Connected((sender, _)) = &self.state {
+                sender
+            } else {
+                return Err(anyhow::anyhow!("Sent into disconnected channel"));
+            };
+            sender
+                .send(ProtoSerializableTimestamp::compute_command_into_proto(cmd))
+                .await
+                .expect("Ok");
+            Ok(())
+        }
 
+        async fn recv(&mut self) -> Result<Option<ComputeResponse<T>>, anyhow::Error> {
+            if let GrpcComputedTcpConn::Connected(channels) = &mut self.state {
+                match channels.1.next().await {
+                    Some(Ok(x)) => Ok(Some(
+                        ProtoSerializableTimestamp::compute_response_into_rust(x)?,
+                    )),
+                    other => {
+                        match other {
+                            Some(Ok(_)) => unreachable!("handled above"),
+                            None => error!("connection unexpectedly terminated cleanly"),
+                            Some(Err(e)) => error!("connection unexpectedly errored: {}", e),
+                        }
+
+                        self.state = GrpcComputedTcpConn::Disconnected;
+                        Err(anyhow::anyhow!("Connection severed"))
+                    }
+                }
+            } else {
+                Err(anyhow::anyhow!("Connection severed"))
+            }
+        }
+    }
+
+    #[async_trait]
+    impl Reconnect for GrpcComputedClient {
+        fn disconnect(&mut self) {
+            info!("GrpcComputedClient: disconnect called");
+            self.state = GrpcComputedTcpConn::Disconnected;
+        }
+
+        async fn reconnect(&mut self) {
+            info!("GrpcComputedClient: reconnect called");
+            self.connect().await
+        }
+    }
+
+    /// A client to a remote dataflow server.
+    ///
+    /// If the client experiences errors, it will attempt a reconnection in the `recv` method and
+    /// produce an error for the call in which that reconnection happens, allowing a bearer to
+    /// re-issue commands. As the reconnection happens in `recv()`, the bearer is advised to use
+    /// a `select` style construct to avoid suspending their task by a call to `recv()`.
```

Comment:
Intentional move of doc comment?

Reply:
I'll add comments. This comment is outdated anyway, TcpClient does not reconnect in recv...